.github/workflows/build.yml

```diff
@@ -8,6 +8,7 @@ jobs:
     strategy:
       matrix:
         platform: [posix, s32k148]
+        cpp-standard: [14, 17, 20]
```

Comment:
this will of course extend our build times. Would probably be enough to just run the compiler front end without generating code. We can use the compiler’s syntax-checking mode or run only the front-end to parse by using the `-fsyntax-only` flag.

Reply:
I don't think ``-fsyntax-only`` is supported by CMake. It would probably require a bigger hack. I tried it naively by adding ``-DCMAKE_C_FLAGS`` which resulted in CMake complaining about a broken compiler which doesn't produce binaries.

We could add ccache in the future. This could help bring down compilation times. Somebody already created a [ccache-action](https://github.com/hendrikmuhs/ccache-action) which combines ccache with github's cache.